src/models.rs

```diff
@@ -0,0 +1,361 @@
+use {
+    cut_optimizer_2d::{CutPiece, PatternDirection, Optimizer, Solution, StockPiece},
+    pdf_canvas::{graphicsstate::Color, BuiltinFont, Pdf},
+    serde::{Deserialize, Serialize},
+    std::{env, fs},
+    cascade::cascade,
+    uom::si::{
+        f32::Length,
+        length::{foot, inch, millimeter},
+    },
+};
+
+pub fn create_solution_pdf(random_seed: u64, solution: Solution) {
+    let mut document =
+        Pdf::create(&format!("solution_{random_seed}.pdf")).expect("Create pdf file");
+    let pt = 2.834;
+    let mut text_output = Vec::new();
+    let (mut doc_width, mut doc_lenght, mut x_os, mut y_os);
+    doc_width = 595.0;
+    doc_lenght = 842.0;
+    let mut stp_n = 1;
+    for stp in solution.stock_pieces {
+        text_output.push(format!("Stockpiece {stp_n}:"));
+        text_output.push("----------------".to_string());
+        if stp.width > stp.length {
+            doc_lenght = 595.0;
+            doc_width = 842.0;
+        }
+        let f = scale_fac(
+            doc_width,
+            doc_lenght,
+            stp.width as f32 * pt,
+            stp.length as f32 * pt,
+        );
+        y_os = (doc_lenght - ((stp.length as f32 * pt) / f)) / 2.0;
+        x_os = (doc_width - ((stp.width as f32 * pt) / f)) / 2.0;
+        document
+            .render_page(doc_width, doc_lenght, |canvas| {
+                canvas.set_stroke_color(Color::rgb(0, 0, 248))?;
+                canvas.set_fill_color(Color::gray(128))?;
+                canvas.rectangle(
+                    x_os,
+                    y_os,
+                    (stp.width as f32 * pt) / f,
+                    (stp.length as f32 * pt) / f,
+                )?;
+                canvas.fill()?;
+                canvas.stroke()?;
+                canvas.rectangle(
+                    x_os,
+                    y_os,
+                    (stp.width as f32 * pt) / f,
+                    (stp.length as f32 * pt) / f,
+                )?;
+                canvas.close_and_stroke()?;
+                stp_n += 1;
+                for cutp in stp.cut_pieces {
+                    text_output.push(format!(
+                        "Id{}: {} x {}mm",
```

Comment:
In the output, there is still "mm", we should output different units.
I will add a patch.